src/Utils/ExpressionsUtil/useGoTo.ts

```diff
@@ -12,38 +12,34 @@ import { passes } from './expression'
 export const useGoTo = (
   data: any,
   onGoTo: (targetPassageId: string) => void,
-): ((name: string) => void) => {
-  const { store } = React.useContext(StoreContext)
-  const { track } = React.useContext(ApiContext)
-  const externalRedirectContext = React.useContext(ExternalRedirectContext)
-  const [goTo, setGoTo] = React.useState<string | null>(null)
-  const { setValue } = React.useContext(StoreContext)
-
-  React.useEffect(() => {
+) => {
+  const { store, setValue } = useContext(StoreContext)
+  const { track } = useContext(ApiContext)
+  const externalRedirectContext = useContext(ExternalRedirectContext)
+
+  const [goTo, setGoTo] = useState<string | null>(null)
+
+  useEffect(() => {
     if (goTo) {
-      const newPassage = data.passages.filter(
+      const newPassage = data.passages.find(
         (passage: any) => passage.name == goTo,
-      )[0]
-      const targetPassage = newPassage ? newPassage.id : data.startPassage
+      )
+      const targetPassageId = newPassage ? newPassage.id : data.startPassage
 
       if (newPassage.redirects.length > 0) {
-        const passableExpressions = newPassage.redirects.filter(
-          (expression: any) => {
-            return passes(store, expression)
-          },
+        const passableExpressions = newPassage.redirects.find(
```

Comment:
Rename this variable since it's not an array anymore

```suggestion
        const passingExpression = newPassage.redirects.find(
```